### PR TITLE
Drop Plotly.plot from the API

### PR DESCRIPTION
**Pull request by archmoj**
Addressing #5395.

TODO: 
 - [x] fix `CI` image test configuration to call `Plotly.newPlot` instead of `Plotly.plot`
 - [x] maintain performance on the CI
 
@plotly/plotly_js

### CODE REVIEW

File: src/plot_api/plot_api.js [hunk 1]
```diff
@@ -552,7 +552,7 @@ function redraw(gd) {
     helpers.cleanLayout(gd.layout);
 
     gd.calcdata = undefined;
-    return exports.plot(gd).then(function() {
+    return exports._doPlot(gd).then(function() {
```
Comment (alexcjohnson) [inline src/plot_api/plot_api.js:555]: Can't we drop `exports.` from these calls in this file?
```suggestion
    return _doPlot(gd).then(function() {
```

Comment (archmoj) [reply src/plot_api/plot_api.js:555]: I did try that.
But some tests failed.

Comment (alexcjohnson) [reply src/plot_api/plot_api.js:555]: oh maybe the spies don't work then. OK, we can leave it.

**Comment by antoinerg** [comment]
> [ ] fix `CI` image test configuration to call `Plotly.newPlot` instead of `Plotly.plot`

Done in https://github.com/plotly/plotly.js/commit/d688bba3be46ded18e25a190b26a6459814ee5ca . ATM I can't really comment on stability / flakiness compared to what we had before.

**Review by alexcjohnson — APPROVED** [review]
💃 Excellent, glad we finally got that to work!

**Pull request merged**